include/zephyr/drivers/firmware/nrf_ironside/cpuconf.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_DRIVERS_FIRMWARE_NRF_IRONSIDE_CPUCONF_H_
+#define ZEPHYR_INCLUDE_ZEPHYR_DRIVERS_FIRMWARE_NRF_IRONSIDE_CPUCONF_H_
+
+#include <zephyr/drivers/firmware/nrf_ironside/call.h>
+
+#include <nrfx.h>
+
+/**
+ * @name CPUCONF service error codes.
+ * @{
+ */
+
+#define IRONSIDE_CPUCONF_ERROR_STARTING_CPU_FAILED      0x5eb0
+#define IRONSIDE_CPUCONF_ERROR_WRONG_CPU                0x5eb2
+#define IRONSIDE_CPUCONF_ERROR_MEM_ACCESS_NOT_PERMITTED 0x5eb1
+#define IRONSIDE_CPUCONF_ERROR_IRONSIDE_CALL_FAILED     0x5eb3
+
+/**
+ * @}
+ */
+
+#define IRONSIDE_CALL_ID_CPUCONF_V0 2
+
+enum {
+	IRONSIDE_CPUCONF_SERVICE_CPU_IDX,
+	IRONSIDE_CPUCONF_SERVICE_VECTOR_TABLE_IDX,
+	IRONSIDE_CPUCONF_SERVICE_CPU_WAIT_IDX,
+	IRONSIDE_CPUCONF_SERVICE_MSG_IDX,
+	IRONSIDE_CPUCONF_SERVICE_MSG_SIZE_IDX,
+	/* The last enum value is reserved for the number of arguments */
+	IRONSIDE_SE_CPUCONF_NUM_ARGS
+};
+
+/* IDX 0 is re-used by the error return code and the 'cpu' parameter. */
+#define IRONSIDE_CPUCONF_SERVICE_RETCODE_IDX 0
```

Comment:
I wonder if we should just define a pattern, like one enum for request indexes and one for response indexes

Reply:
I'd rather not diverge from the update service.

But this sounds like a nice refactoring for when the update service has been merged.